lib/index.js

```diff
@@ -66,28 +54,38 @@ const plugin = {
         // Deprecated rules.
         "no-hide-core-modules": require("./rules/no-hide-core-modules"),
         shebang: require("./rules/shebang"),
-    }),
-    configs: {
-        "recommended-module": { plugins: ["n"], ...esmConfig.eslintrc },
-        "recommended-script": { plugins: ["n"], ...cjsConfig.eslintrc },
-        recommended: { plugins: ["n"], ...recommendedConfig.eslintrc },
-        "flat/recommended-module": { ...esmConfig.flat },
-        "flat/recommended-script": { ...cjsConfig.flat },
-        "flat/recommended": { ...recommendedConfig.flat },
-        "flat/mixed-esm-and-cjs": [
-            { files: ["**/*.js"], ...recommendedConfig.flat },
-            { files: ["**/*.mjs"], ...esmConfig.flat },
-            { files: ["**/*.cjs"], ...cjsConfig.flat },
-        ],
     },
 }
+/**
+ * @typedef {{
+ *     'recommended-module': import('eslint').ESLint.ConfigData;
+ *     'recommended-script': import('eslint').ESLint.ConfigData;
+ *     'recommended': import('eslint').ESLint.ConfigData;
+ *     'flat/recommended-module': import('eslint').Linter.FlatConfig;
+ *     'flat/recommended-script': import('eslint').Linter.FlatConfig;
+ *     'flat/recommended': import('eslint').Linter.FlatConfig;
+ *     'flat/mixed-esm-and-cjs': import('eslint').Linter.FlatConfig[];
+ * }} Configs
```

Comment:
If you want it a bit shorter:

```suggestion
 * @typedef {
 *      Record<'recommended-module' | 'recommended-script' | 'recommended',  import('eslint').ESLint.ConfigData> & 
 *      Record<'flat/recommended-module' | 'flat/recommended-script' | 'flat/recommended',  import('eslint').ESLint.FlatConfig> & 
 *      Record<'flat/mixed-esm-and-cjs', import('eslint').Linter.FlatConfig[]>
 * }} Configs
```

Or even short, but less searchable:

```suggestion
 * @typedef {
 *      Record<`recommended${'-module' | '-script' | ''}`,  import('eslint').ESLint.ConfigData> & 
 *      Record<`flat/recommended${'-module' | '-script' | ''}`,  import('eslint').ESLint.FlatConfig> & 
 *      Record<'flat/mixed-esm-and-cjs', import('eslint').Linter.FlatConfig[]>
 * }} Configs
```

Reply:
hehe, I tried a few different combinations, if I could I would have used:
```js
/**
 * @typedef Configs
 * @prop {import('eslint').ESLint.ConfigData}  recommended-module
 * @prop {import('eslint').ESLint.ConfigData}  recommended-script
 * @prop {import('eslint').ESLint.ConfigData}  recommended
 * @prop {import('eslint').Linter.FlatConfig}  flat/recommended-module
 * @prop {import('eslint').Linter.FlatConfig}  flat/recommended-script
 * @prop {import('eslint').Linter.FlatConfig}  flat/recommended
 * @prop {import('eslint').Linter.FlatConfig[]}  flat/mixed-esm-and-cjs
 */
```

My version of your suggestions was:
```js
/**
 * @typedef {(
 *   & Record<
 *       | 'recommended-module'
 *       | 'recommended-script'
 *       | 'recommended',
 *     import('eslint').ESLint.ConfigData
 *   >
 *   & Record<
 *       | 'flat/recommended-module'
 *       | 'flat/recommended-script'
 *       | 'flat/recommended',
 *     import('eslint').ESLint.FlatConfig
 *   >
 *   & Record<
 *     'flat/mixed-esm-and-cjs',
 *     import('eslint').Linter.FlatConfig[]
 *   >
 * )} Configs
 */
```

I ended settling on whats in this PR as I prefer the verbosity :thinking:

Happy to change if its prefered though :joy: 

---

All that being said, I think in the next major release we should go down to:

```js
/**
 * @typedef Configs
 * @prop {import('eslint').Linter.FlatConfig}  recommended
 * @prop {import('eslint').Linter.FlatConfig}  recommended-module
 * @prop {import('eslint').Linter.FlatConfig}  recommended-script
 * @prop {import('eslint').Linter.FlatConfig[]}  recommended-mixed
 */
```

Reply:
Could possibly use the new `/** @import { foo } from 'bar' */` then as well, to avoid having the ESLint import in multiple places

Reply:
<img width="441" alt="image" src="https://github.com/user-attachments/assets/623be1a2-5ec6-4a87-9cf7-4405f1ae7261">